x-pack/plugins/siem/public/components/flow_controls/flow_target_select_connected.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexItem } from '@elastic/eui';
+import React from 'react';
+import { connect } from 'react-redux';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+import { ActionCreator } from 'typescript-fsa';
+
+import { FlowDirection, FlowTarget } from '../../graphql/types';
+import { State } from '../../store';
+import { networkActions, networkSelectors } from '../../store/network';
+import { IpOverviewId } from '../page/network/ip_overview';
+import * as i18nIp from '../page/network/ip_overview/translations';
+
+import { FlowTargetSelect } from './flow_target_select';
+
+const SelectTypeItem = styled(EuiFlexItem)`
+  min-width: 180px;
+`;
+
+interface FlowTargetSelectReduxProps {
+  flowTarget: FlowTarget;
+}
+
+export interface FlowTargetSelectDispatchProps {
+  updateIpDetailsFlowTarget: ActionCreator<{
```

Comment:
Just a question for your thoughts around the `FlowTargetSelectConnected` component. I assume you extracted a connected component to simplify things in the `PageHeadlineComponent`? That seems 👍, but now `FlowTargetSelectConnected` looks to be coupled to the `updateIpDetailsFlowTarget` action.

Is this connected component only intended for use within `IP Details`, or to be a more generic connected version of `FlowTargetSelect`? If the former, it may make sense to move it to the `.../components/page/network/` package since it wouldn't be a generic component that could be used on any page, if the latter, it may make sense to restructure this component such that the connected action can be specified, otherwise if this component is used on a page outside `IP Details`, it would then call `networkActions.updateIpDetailsFlowTarget` and update the `IP Details` redux state.

Reply:
I'll move it to the page components, thanks